src/marqo/tensor_search/tensor_search.py

```diff
@@ -856,16 +860,21 @@ def _add_documents_structured(config: Config, add_docs_params: AddDocsParams, ma
                                 chunk_id = f"{chunk_time}"
                                 chunks.append(chunk_id)
 
-                                vector = s2_inference.vectorise(
-                                    model_name=marqo_index.model.name,
-                                    content=[media_chunk['tensor']],  # Wrap in list as vectorise expects an iterable
-                                    model_properties=marqo_index.model.get_properties(),
-                                    device=add_docs_params.device,
-                                    normalize_embeddings=marqo_index.normalize_embeddings,
-                                    infer=True,
-                                    model_auth=add_docs_params.model_auth,
-                                    modality=Modality.VIDEO if marqo_field.type == FieldType.VideoPointer else Modality.AUDIO
-                                )
+                                start_time = timer()
+                                with RequestMetricsStore.for_request().time(f"add_documents.create_vectors"):
+                                    vector = s2_inference.vectorise(
+                                        model_name=marqo_index.model.name,
+                                        content=[media_chunk['tensor']],  # Wrap in list as vectorise expects an iterable
+                                        model_properties=marqo_index.model.get_properties(),
+                                        device=add_docs_params.device,
+                                        normalize_embeddings=marqo_index.normalize_embeddings,
+                                        infer=True,
+                                        model_auth=add_docs_params.model_auth,
+                                        modality=Modality.VIDEO if marqo_field.type == FieldType.VideoPointer else Modality.AUDIO
+                                    )
+
+                                end_time = timer()
+                                total_vectorise_time += (end_time - start_time)
```

Comment:
Why is this calculated but not used?

Reply:
It's used in 

```logger.debug(f"          add_documents vectorise: took {(total_vectorise_time):.3f}s for {batch_size} docs, "
                 f"for an average of {(total_vectorise_time / batch_size):.3f}s per doc.")```

Reply:
The diff says you're adding this variable, but there's no diff to show it's being used. Was the line using it erroring out before?

Reply:
Nope, the variable already existed before, we just wrapped the audio/video vectorise call in this and added their times to the total as well.